numpy/_array_api/_creation_functions.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import annotations
+
+from ._types import (Optional, SupportsDLPack, SupportsBufferProtocol, Tuple,
+                     Union, array, device, dtype)
+from ._dtypes import _all_dtypes
+
+import numpy as np
+
+def asarray(obj: Union[float, NestedSequence[bool|int|float], SupportsDLPack, SupportsBufferProtocol], /, *, dtype: Optional[dtype] = None, device: Optional[device] = None, copy: Optional[bool] = None) -> array:
+    """
+    Array API compatible wrapper for :py:func:`np.asarray <numpy.asarray>`.
+
+    See its docstring for more information.
+    """
+    # _array_object imports in this file are inside the functions to avoid
+    # circular imports
+    from ._array_object import ndarray
+    if device is not None:
+        # Note: Device support is not yet implemented on ndarray
+        raise NotImplementedError("Device support is not yet implemented")
+    if copy is not None:
+        # Note: copy is not yet implemented in np.asarray
+        raise NotImplementedError("The copy keyword argument to asarray is not yet implemented")
+    if isinstance(obj, ndarray):
+        return obj
```

Comment:
This surprises me a little, as this means `asarray(int_arr, dtype=float)` in the array API returns an int array, not a float array. I don't see this behavior specified at https://data-apis.github.io/array-api/latest/API_specification/creation_functions.html#asarray-obj-dtype-none-device-none-copy-none.

Reply:
That's just a bug on my part. I should also note that this code hasn't been fully reviewed yet, and only a subset of the test suite that would catch these kinds of bugs is implemented so far (tests for asarray aren't implemented yet).